rest/indextest/index_init_api_test.go

```diff
@@ -134,15 +156,29 @@ func TestChangeIndexPartitions(t *testing.T) {
 			}
 			dbConfig.Index.NumPartitions = base.Ptr(test.newPartitions)
 			rest.RequireStatus(t, rt.ReplaceDbConfig(dbName, dbConfig), http.StatusCreated)
+			require.Equal(t, test.newPartitions, *rt.GetDatabase().Options.NumIndexPartitions)
 
 			// CBG-4565 cleanup old indexes
-			//resp = rt.SendAdminRequest(http.MethodPost, "/_post_upgrade", "")
-			//rest.RequireStatus(t, resp, http.StatusOK)
-			//var body rest.PostUpgradeResponse
-			//err := base.JSONUnmarshal(resp.BodyBytes(), &body)
-			//require.NoError(t, err)
-			//require.Lenf(t, body.Result, 1, "expected one database in post upgrade response")
-			//require.Lenf(t, body.Result[dbName].RemovedIndexes, 2, "expected two indexes to be removed")
+			resp = rt.SendAdminRequest(http.MethodPost, "/_post_upgrade", "")
+			rest.RequireStatus(t, resp, http.StatusOK)
+			var body rest.PostUpgradeResponse
+			err := base.JSONUnmarshal(resp.BodyBytes(), &body)
+			collection := db.GetSingleDatabaseCollection(t, database)
+			var removedIndexes []string
+			for _, index := range test.removedIndexes {
+				removedIndexes = append(removedIndexes, fmt.Sprintf("`%s`.`%s`.%s", collection.ScopeName, collection.Name, index))
+			}
+
+			expectedOutput := rest.PostUpgradeResponse{
+				Result: rest.PostUpgradeResult{
+					"db": rest.PostUpgradeDatabaseResult{
+						RemovedDDocs:   []string{},
+						RemovedIndexes: removedIndexes,
+					},
+				},
+			}
+			require.NoError(t, err)
+			require.Equal(t, expectedOutput, body)
```

Comment:
Do we have any tests that ensure we're not removing non-SGW indexes, if they exist?